add CleanWebpackPlugin

peterbe opened the pull request:
Fixes #240

1. `yarn workspace cli build`
2. `touch cli/dist/junk`
3. `yarn workspace cli build` (again)
4. 
```
> ls cli/dist
cli.js     cli.js.map
```
See, the `junk` file is removed.

peterbe commented:
What the heck! Why is CI saying the lock file is corrupt? Here's how I added it:
```
yarn workspace cli add --dev clean-webpack-plugin
```
Is that now how you use `yarn add` with workspaces?

Gregoor commented:
So strange, I can't reproduce this with a minimal workspace repo, might be a fairly specific bug. I have a bit of a schtupid-fix for now: We put `yarn install --ignore-scripts` into our pre-commit hooks. When the lockfile is up-to-date this should be a sub-second operation.

peterbe commented:
@Gregoor does this not happen for you on your laptop too? Simply going to `master` to this branch and running `yarn` causes changes to `yarn.lock`:

```
~/stumptown-renderer  master ✔
▶ gg branches peterbe:add-cleanwebpackplugin
Found existing branches...
--------------------------------------------------------------------------------
add-cleanwebpackplugin
	2019-11-19T21:10:03
	2 days, 18:11:17.871047
	commit (amend): add CleanWebpackPlugin

Check out 'add-cleanwebpackplugin'? [Y/n]

~/stumptown-renderer  add-cleanwebpackplugin ✔
▶ yarn
yarn install v1.19.1
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
[3/5] 🚚  Fetching packages...
[4/5] 🔗  Linking dependencies...
warning "workspace-aggregator-adf26906-ee43-4ea4-9f3a-8a90f398109a > client > react-scripts > @typescript-eslint/eslint-plugin > tsutils@3.17.1" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
[5/5] 🔨  Building fresh packages...
success Saved lockfile.
$ cd stumptown && npm install && git checkout package-lock.json
npm WARN sinon-as-promised@4.0.3 requires a peer of sinon@1 but none is installed. You must install peer dependencies yourself.

audited 1282 packages in 3.521s
found 3 vulnerabilities (2 moderate, 1 high)
  run `npm audit fix` to fix them, or `npm audit` for details
Updated 1 path from the index
✨  Done in 13.45s.

~/stumptown-renderer  add-cleanwebpackplugin ✗
▶ gst
On branch add-cleanwebpackplugin
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   yarn.lock

no changes added to commit (use "git add" and/or "git commit -a")
```

The diff always becomes: https://gist.github.com/peterbe/5da2a265b33620c9c7521bfe40c8f9b5

Are you not getting the same on your laptop?

Gregoor commented:
yup, I am getting the same. And I can also reproduce it completely (i.e. new brach -> yarn add plugin -> commit -> yarn -> status shows changes). My idea for working around it is to add a pre-commit hook which always runs yarn (super quick when nothing has changed).

peterbe commented:
But a hook us personal. What is the command you need to run to fix things
if you genuinely need to yarn add something today?
We could put that it the README for now.

On Mon, Nov 25, 2019 at 7:53 AM Gregor <notifications@github.com> wrote:

> yup, I am getting the same. And I can also reproduce it completely (i.e.
> new brach -> yarn add plugin -> commit -> yarn -> status shows changes). My
> idea for working around it is to add a pre-commit hook which always runs
> yarn (super quick when nothing has changed).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mdn/stumptown-renderer/pull/241?email_source=notifications&email_token=AAAGQ46JMAXX4FQUVPSMQMDQVPDEZA5CNFSM4JPJODG2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFCJGRA#issuecomment-558142276>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAGQ46X677MZTBN5X6UZN3QVPDEZANCNFSM4JPJODGQ>
> .
>
-- 
Peter Bengtsson
Mozilla MDN Web Docs
https://www.peterbe.com

Gregoor commented:
Oh I would've used husky to make sure everyone gets that hook, then we could forego the README, no?

peterbe commented:
> Oh I would've used husky to make sure everyone gets that hook, then we could forego the README, no?

I'd rather not because then we're "brushing the problem under the mat" instead of raising awareness of it. Because hopefully this'll get fixed in the coming months.

Gregoor commented:
How about also creating an issue and using [a github reminder bot](https://github.com/probot/reminders) to nudge us in a couple of months and test if a newer yarn version fixes it?

peterbe commented:
Replaced by https://github.com/mdn/stumptown-renderer/pull/297

peterbe commented:
Replaced by https://github.com/mdn/stumptown-renderer/pull/297
